### PR TITLE
[formidable] Add createDirsFromUploads option, fix type for Fields interface and Files interface

### PR DESCRIPTION
**Pull request by martin-badin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test `formidable`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Discussion: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/66218

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## formidable ([<kbd>unpkg</kbd>](https://unpkg.com/browse/formidable@latest/))
was missing the following properties:
1. OctetstreamParser
1. QueryStringParser
1. octetstream
1. querystring
1. multipart

<details>
<summary>as well as these 1 other properties...</summary>
<p>json</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against cc0150ce2fe61b5c2f5c7316492dadde8d10ba74
</p>

File: types/formidable/index.d.ts [hunk 1]
```diff
@@ -184,6 +184,13 @@ declare namespace formidable {
          */
         multiples?: boolean | undefined;
 
+        /**
+         * If true, makes direct folder uploads possible.
+         *
+         * @default false
+         */
+        createDirsFromUploads?: boolean;
```
Comment (peterblazejewicz) [inline types/formidable/index.d.ts:192]:
```suggestion
        createDirsFromUploads?: boolean | undefined;
```
(to stick with already applied convention) thx!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@martin-badin thanks!

**Comment by martin-badin** [comment]
Ready to merge

**Pull request merged**